contracts/FIFOMarket.sol

```diff
@@ -65,25 +77,53 @@ contract FIFOMarket is
     );
     _queueHeadIndex = 0;
     _queueNextInsertIndex = 0;
+    priorityRestrictedThreshold = 0;
+    totalSupply = 0;
+    _grantRole(DEFAULT_ADMIN_ROLE, _msgSender());
+    _grantRole(ALLOWLIST_ROLE, _msgSender()); // todo, do this here or have to do it explicitly after intialization?
+  }
+
+  function setPriorityRestrictedThreshold(uint256 threshold)
+    external
+    onlyRole(DEFAULT_ADMIN_ROLE)
+  {
+    priorityRestrictedThreshold = threshold;
+    emit PriorityRestrictedThresholdSet(threshold);
   }
 
   function _queueLength() private view returns (uint256) {
     return _queueNextInsertIndex - _queueHeadIndex;
   }
 
-  // todo do we want this to report NRTs in either or wei? because removal token balances are denominated in wei
-  function numberOfNrtsInQueue() public view returns (uint256) {
+  function numberOfNrtsInQueueComputed() public view returns (uint256) {
     uint256 nrtsInQueue = 0;
     for (uint256 i = _queueHeadIndex; i < _queueNextInsertIndex; i++) {
       nrtsInQueue += _removal.balanceOf(address(this), _queue[i]);
     }
     return nrtsInQueue;
   }
 
-  function nextRemovalForSale() public view returns (uint256) {
+  function totalUnrestrictedSupply() public view returns (uint256) {
```

Comment:
Any reason you opted for a `totalSupply` class variable, instead of just reusing `numberOfNrtsInQueue`?
Such that `totalUnrestrictedSupply = numberOfNrtsInQueue - priorityRestrictedTreshold`

Reply:
that `totalSupply` class variable is necessary for other reasons and provides the same information as `numberOfNrtsInQueueComputed`.  I figure using it here makes it so that we don't have to change anything if we ever call this method internally in such a way that the efficiency matters.